### PR TITLE
Add GetAllRecords

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/NameService/NameService.cs [hunk 1]
```diff
@@ -301,7 +306,19 @@ public static string GetRecord(string name, RecordType type)
             NameState token = (NameState)StdLib.Deserialize(nameMap[tokenKey]);
             token.EnsureNotExpired();
             byte[] recordKey = GetRecordKey(tokenKey, name, type);
-            return recordMap[recordKey];
+            return recordMap.Get<RecordState>(recordKey).Data;
```
Comment (cloud8little) [inline src/NameService/NameService.cs]:
```suggestion
            return ((RecordState)StdLib.Deserialize(recordMap.Get(recordKey))).Data;
```

File: src/NameService/NameService.cs [hunk 2]
```diff
@@ -285,7 +285,12 @@ public static void SetRecord(string name, RecordType type, string data)
             token.EnsureNotExpired();
             token.CheckAdmin();
             byte[] recordKey = GetRecordKey(tokenKey, name, type);
-            recordMap[recordKey] = data;
+            recordMap.Put(recordKey, new RecordState
+            {
+                Name = name,
+                Type = type,
+                Data = data
+            });
```
Comment (cloud8little) [inline src/NameService/NameService.cs:293]:
```suggestion
                       recordMap.Put(recordKey, StdLib.Serialize(new RecordState
            {
                Name = name,
                Type = type,
                Data = data
            }));
```

File: src/NameService/NameService.cs (hunk 1 above)
Comment (erikzhang) [reply src/NameService/NameService.cs]: What's the difference?

File: src/NameService/NameService.cs (hunk 2 above)
Comment (erikzhang) [reply src/NameService/NameService.cs:293]: What's the difference?

Comment (cloud8little) [reply src/NameService/NameService.cs:293]: > What's the difference?

the framework compile will show error

**Comment by cloud8little** [comment]
Test passed with IPV6 set.

**Review by cloud8little — APPROVED** [review]

**Pull request merged**